rest_framework/permissions.py

```diff
@@ -78,8 +78,8 @@ def has_permission(self, request, view):
 
     def has_object_permission(self, request, view, obj):
         return (
-            self.op1.has_object_permission(request, view, obj) or
-            self.op2.has_object_permission(request, view, obj)
+            (self.op1.has_permission(request, view) and self.op1.has_object_permission(request, view, obj)) or
+            (self.op2.has_permission(request, view) and self.op2.has_object_permission(request, view, obj))
         )
```

Comment:
return (
            self.op1.has_permission(request, view)
            and self.op1.has_object_permission(request, view, obj)
        ) or (
            self.op2.has_permission(request, view)
            and self.op2.has_object_permission(request, view, obj)
        )